dev/Repeater/SelectionModel.cpp

```diff
@@ -580,18 +580,15 @@ void SelectionModel::ClearSelection(bool resetAnchor, bool raiseSelectionChanged
 
 void SelectionModel::OnSelectionChanged()
 {
+    auto const previousSelectedIndicies = winrt::IVectorView<winrt::IndexPath>(m_selectedIndicesCached);
+
     m_selectedIndicesCached = nullptr;
     m_selectedItemsCached = nullptr;
 
     // Raise SelectionChanged event
     if (m_selectionChangedEventSource)
     {
-        if (!m_selectionChangedEventArgs)
-        {
-            m_selectionChangedEventArgs = tracker_ref<winrt::SelectionModelSelectionChangedEventArgs>(this, winrt::make<SelectionModelSelectionChangedEventArgs>());
-        }
-
-        m_selectionChangedEventSource(*this, m_selectionChangedEventArgs.get());
+        m_selectionChangedEventSource(*this, SelectionModelSelectionChangedEventArgs(previousSelectedIndicies));
```

Comment:
We used to cache the args object and not allocate every time. Is there a reason to not do that anymore.

Reply:
The args contains a previous selected indices vector now instead of being an empty args that could be freely reused. I suppose we could have 1 args object which gets its vector updated.  That could be weird though if someone who receives the args object changes the vector in it. 

---
In reply to: [347404996](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347404996) [](ancestors = 347404996)

Reply:
Removed the change to selection model

---
In reply to: [347522163](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347522163) [](ancestors = 347522163,347404996)